### ISSUE TITLE
Error: Class "Keyword" not found

### ISSUE THREAD

**Issue by jonyking** [googleads/googleads-php-lib#551]
Hello,

I get the error message: Fatal error: Uncaught Error: Class 'Keyword' not found ..

Code:

```
<?php

require_once(__DIR__ . '/vendor/autoload.php');
 

// Create keywords. Up to 2000 keywords can be passed in a single request.
$keywords = [];

$keyword = new Keyword();
$keyword->setText('mars cruise');
$keyword->setMatchType(KeywordMatchType::BROAD);
$keywords[] = $keyword;

$keyword = new Keyword();
$keyword->setText('cheap cruise');
$keyword->setMatchType(KeywordMatchType::PHRASE);
$keywords[] = $keyword;

$keyword = new Keyword();
$keyword->setText('cruise');
$keyword->setMatchType(KeywordMatchType::EXACT);
$keywords[] = $keyword;

// Create a keyword estimate request for each keyword.
$keywordEstimateRequests = [];
foreach ($keywords as $keyword) {
    $keywordEstimateRequest = new KeywordEstimateRequest();
    $keywordEstimateRequest->setKeyword($keyword);
    $keywordEstimateRequests[] = $keywordEstimateRequest;
}

// Negative keywords don't return estimates, but adjust the estimates of the
// other keywords in the hypothetical ad group.
$negativeKeywords = [];

$keyword = new Keyword();
$keyword->setText('moon walk');
$keyword->setMatchType(KeywordMatchType::BROAD);
$negativeKeywords[] = $keyword;

// Create a keyword estimate request for each negative keyword.
foreach ($negativeKeywords as $negativeKeyword) {
    $keywordEstimateRequest = new KeywordEstimateRequest();
    $keywordEstimateRequest->setKeyword($negativeKeyword);
    $keywordEstimateRequest->setIsNegative(true);
    $keywordEstimateRequests[] = $keywordEstimateRequest;
}

// Create ad group estimate requests.
$adGroupEstimateRequest = new AdGroupEstimateRequest();
$adGroupEstimateRequest->setKeywordEstimateRequests(
    $keywordEstimateRequests
);
$money = new Money();
$money->setMicroAmount(1000000);
$adGroupEstimateRequest->setMaxCpc($money);

// Create campaign estimate requests.
$campaignEstimateRequest = new CampaignEstimateRequest();
$campaignEstimateRequest->setAdGroupEstimateRequests(
    [$adGroupEstimateRequest]
);

// Optional: Set additional criteria for filtering estimates.
// See http://code.google.com/apis/adwords/docs/appendix/countrycodes.html
// for a detailed list of country codes.
// Set targeting criteria. Only locations and languages are supported.
$unitedStates = new Location();
$unitedStates->setId(2840);

// See http://code.google.com/apis/adwords/docs/appendix/languagecodes.html
// for a detailed list of language codes.
$english = new Language();
$english->setId(1000);

$campaignEstimateRequest->setCriteria([$unitedStates, $english]);

// Create selector.
$selector = new TrafficEstimatorSelector();
$selector->setCampaignEstimateRequests([$campaignEstimateRequest]);

// Optional: Request a list of campaign level estimates segmented by
// platform.
$selector->setPlatformEstimateRequested(true);

$result = $trafficEstimatorService->get($selector);

$platformEstimates = $result->getCampaignEstimates()[0]->getPlatformEstimates();
if ($platformEstimates !== null) {
    foreach ($platformEstimates as $platformEstimate) {
        if ($platformEstimate->getMinEstimate() !== null
            && $platformEstimate->getMaxEstimate() !== null) {
            printf(
                "Results for the platform with ID %d and name '%s':\n",
                $platformEstimate->getPlatform()->getId(),
                $platformEstimate->getPlatform()->getPlatformName()
            );
            self::printMeanEstimate(
                $platformEstimate->getMinEstimate(),
                $platformEstimate->getMaxEstimate()
            );
        }
    }
}

$keywordEstimates = $result->getCampaignEstimates()[0]->getAdGroupEstimates()[0]->getKeywordEstimates();
$estimatesCount = count($keywordEstimates);
for ($i = 0; $i < $estimatesCount; $i++) {
    $keywordEstimateRequest = $keywordEstimateRequests[$i];
    // Skip negative keywords, since they don't return estimates.
    if ($keywordEstimateRequest->getIsNegative() !== true) {
        $keyword = $keywordEstimateRequest->getKeyword();
        $keywordEstimate = $keywordEstimates[$i];

        if ($keywordEstimate->getMin() !== null
            && $keywordEstimate->getMax() !== null) {
            // Print the mean of the min and max values.
            printf(
                "Results for the keyword with text '%s' and match type '%s':\n",
                $keyword->getText(),
                $keyword->getMatchType()
            );
            self::printMeanEstimate(
                $keywordEstimate->getMin(),
                $keywordEstimate->getMax()
            );
        }
    }
}

```

**Comment by fiboknacky** [comment]
Hello,

What library version (not API version) are you using?
And can you make sure that the path to your autoload.php file is correct please?

Best,
Knack

**Comment by fiboknacky** [comment]
Oh, it's just that you've not import any classes here yet.
See [GetKeywords.php](https://github.com/googleads/googleads-php-lib/blob/master/examples/AdWords/v201809/BasicOperations/AddKeywords.php) for details on how to import classes:

```php
use Google\AdsApi\AdWords\v201809\cm\Keyword;
```

**Comment by jonyking** [comment]
Not working. Same issue.

**Comment by fiboknacky** [comment]
Can you run our [GetKeywords](https://github.com/googleads/googleads-php-lib/blob/master/examples/AdWords/v201809/BasicOperations/AddKeywords.php) example without any errors?

If so, can you adapt from that example instead? It would be much easier than starting from scratch.

**Comment by jonyking** [comment]
There I got the following error:
`Fatal error: Uncaught InvalidArgumentException: All of 'clientId', 'clientSecret', and 'refreshToken' must be set when using installed/web application flow. in ...`

**Comment by fiboknacky** [comment]
You've not set the configuration file properly.
Please go through the [README.md](https://github.com/googleads/googleads-php-lib/blob/master/README.md) file and follow the instructions there.

If you have questions regarding client ID, etc., in general, please reach out to our [AdWords API forum](https://groups.google.com/forum/?hl=de#!forum/adwords-api).

Best,
Knack

**Comment by jonyking** [comment]
I have set up the configuration file exactly how it is described in the file ...

**Comment by fiboknacky** [comment]
So, could you please share your config file?

**Comment by jonyking** [comment]
My `adsapi_php.ini`

```
developerToken = "XXXX"
clientCustomerId = "XXX-XXX-XXXX"

userAgent = "company_name"

clientId = "XXXXXXXXapps.googleusercontent.com"
clientSecret = "XXXXX"
refreshToken = "XXXXXX"
```

**Comment by jonyking** [comment]
The folder has 755 permissions. The .ini file was created locally.

**Comment by fiboknacky** [comment]
Are they under the "[ADWORDS]" and "[OAUTH2]" sections in the file?
It looks like they're not and that's why the library cannot find them.

**Comment by jonyking** [comment]
I do not have these sections. How they will be added? - When creating the .ini file?  Something wrong with the permissions?

**Comment by fiboknacky** [comment]
The permissions are not related to your issue. You just need to follow the format of the config file closely.

And that's why I would like to ask you to start from the beginning again by following README closely. :)
In the [Getting started](https://github.com/googleads/googleads-php-lib#getting-started) section, the instruction is to _copy_ the sample file of `adsapi_php.ini` to your home directory and just fill in your information.
In this way, you will not have a problem with the sections of the file.

Could you please copy the sample file and fill in your information instead?

**Comment by jonyking** [comment]
I have copied the original `adsapi_php.ini` and copied my content to it. Same result.

**Comment by fiboknacky** [comment]
So, do your  `adsapi_php.ini` contain the section header now? Can you post the content (hiding some sensitive info) here?
And the error is still the same?
How did you download and try code examples? Could you please elaborate in detail? 
Thanks.

**Comment by jonyking** [comment]
Ok, so now it is working. Had a little mistake inside de `adsapi_php.ini`. - sorry for that!

Now I got a new error:

`Fatal error: Uncaught Google\AdsApi\AdWords\v201809\cm\ApiException: [AuthenticationError.CUSTOMER_NOT_FOUND @ ; trigger:'<null>'] `

I think the customer ID is wrong?

**Comment by fiboknacky** [comment]
Yes, the customer ID is wrong.

**Comment by jonyking** [comment]
Ok, now I got the error: `DEVELOPER_TOKEN_NOT_APPROVED`

**Comment by fiboknacky** [comment]
I would recommend reading the [Get Started](https://developers.google.com/adwords/api/docs/guides/start) guide before starting using this client library.
As for your information for now, you have to particularly read [this page](https://developers.google.com/adwords/api/docs/guides/signup).

In case you have problems with token approval, please ask in the [AdWords API forum](https://groups.google.com/forum/?hl=EN#!forum/adwords-api).
There are people monitoring the forum so you can get answers quickly.

I'll close this issue for now because it looks like you have an issue with using AdWords API in general, instead of this client library.
Feel free to create a new issue in the future if you have any problems with this client library usage.

Best,
Knack
